File: bench.pl
use strict;
use Time::HiRes qw(time);
use LWP::UserAgent;
use JSON::XS;

my (@body, $body, $json, $ua, $t);

$ua = new LWP::UserAgent;
$ua->request(HTTP::Request->new('GET', 'http://dreadatour.ru/larry.js'), sub { push @body, shift });

$t = time;
$body = join('', @body) foreach 1..100;
printf "join: %dms\n", int((time - $t) * 1000 + 0.5);

$t = time;
$json = JSON::XS->new->ascii->decode($body) foreach 1..100;
printf "parse: %dms\n", int((time - $t) * 1000 + 0.5);

File: bench.js
var i, start, http = require('http'), join = 0, parse = 0;
for (i = 0; i < 100; i++){
http.request({host:'dreadatour.ru', port:80, path:'/larry.js', method:'GET'}, function(response){
    var body = new Array();
    response.setEncoding('utf8');
    response.on('data', function(chunk){body.push(chunk)})
            .on('end', function(){
                start = new Date().getTime();
                body = body.join('');
                join += (new Date().getTime() - start);
                
                start = new Date().getTime();
                JSON.parse(body);
                parse += (new Date().getTime() - start);
                console.log(join, parse);
            });
}).end()
}
